File: C++ Solutions/237. Delete Node in a Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        if(node -> next != NULL){ node -> val = node -> next -> val; }
        if(node -> next -> next == NULL){ node -> next = NULL; return; }
        deleteNode(node -> next);
    }
};

File: Python Solutions/237. Delete Node in a Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, node):     
        if node.next != None: node.val = node.next.val
        if node.next.next == None:
            node.next = None
        if node.next != None: 
            self.deleteNode(node.next)
